packages/big-design/src/components/Worksheet/types.ts

```diff
@@ -0,0 +1,21 @@
+export interface WorksheetColumn {
+  hash: string;
+  header: string;
+  type?: 'text' | 'number';
+  validation?(value: string | number): boolean;
+}
+
+export interface WorksheetProps<T> {
+  columns: WorksheetColumn[];
+  items: T[];
+  onChange(items: Cell[]): void;
+  onErrors?(items: Cell[]): void;
+}
+
+export interface Cell {
+  columnIndex: number;
+  hash: string;
+  rowIndex: number;
+  value: string | number;
```

Comment:
Let's see if we can infer this type through a generic. I took a quick look, and you might explore this:
```ts
interface Product {
    productName: string;
    categories: string;
    otherField: string;
    otherField2: string;
    otherField3: number;
}

interface Worksheet<Item> {
    items: Item[];
    onChange(items: Array<Cell<Item[keyof Item]>>): void;
}

interface Cell<Value> {
    value: Value;
}

const ws: Worksheet<Product> = {
    items: [],
    onChange: items => {
        console.log(items);
    },
};
```
![Screen Shot 2021-04-05 at 16 09 41](https://user-images.githubusercontent.com/10539418/113627599-5fcc9280-9629-11eb-893f-b4663308c638.png)

Reply:
This is super gucci but the problem I see is that I use the `Cell` type in a lot of places. What would be a good solution for that?

Reply:
Yeah, `unknown` would be the best bet. Is there a problem with that approach?